### ISSUE TITLE
Conform to PyTorch convention in the loss

### ISSUE THREAD

**Issue by wiseodd** [aleximmer/Laplace#163]
Currently we always assume that the class index in the outputs of the model to be `-1`. While this works for standard models and Huggingface LLMs, there are some important models where this is false. E.g. image-output models where the logit tensor is of shape `(n_batch, n_classes, height, width)`, i.e. the class index is `1`.

**Comment by wiseodd** [comment]
The current idea is to add an arg in `BaseLaplace`: `logit_class_idx: int = -1`. Then, whenever Laplace flattens logits, it will use that as guidance.

Test cases for conv last layer, which will result in a logit of shape `(batch_size, n_classes, height, width)`, should be created to cover this use case.
